### PR TITLE
config role for function

### LINKED ISSUE #220: Container app targetPort should default to 80, and have 8080 in app bicep files.
The targetPort was changed to 8080 here:

https://github.com/Azure-Samples/azure-search-openai-demo-csharp/blob/fb1ca3248e693ea7acb04740c97526c5a49c82eb/infra/core/host/container-app.bicep#L71

Instead of changing /core, the targetPort should be updated in the application bicep.

This allows you to easily update to latest /core when needed.

### LINKED ISSUE #221: Application insights is disabled by default leading to poor customer debugging experience.
This project was updated to not include application insights by default.  When trying to debug an app the user will need to jump through many hoops to enable it.  I recommend reverting to having application insights turned on by default.

https://github.com/Azure-Samples/azure-search-openai-demo-csharp/blob/main/infra/core/monitor/monitoring.bicep#L2

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

fix #221 
fix #220 

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: infra/main.bicep [hunk 1]
```diff
@@ -634,3 +634,4 @@ output AZURE_STORAGE_RESOURCE_GROUP string = storageResourceGroup.name
 output AZURE_TENANT_ID string = tenant().tenantId
 output SERVICE_WEB_IDENTITY_NAME string = web.outputs.SERVICE_WEB_IDENTITY_NAME
 output SERVICE_WEB_NAME string = web.outputs.SERVICE_WEB_NAME
+output FUNCTION_SERVICE_IDENTITY_PRINCIPAL_ID string = function.outputs.SERVICE_FUNCTION_IDENTITY_PRINCIPAL_ID
```
Comment (jongio) [inline infra/main.bicep:637]: follow SERVICE_FUNCTION pattern in name

File: infra/main.parameters.json [hunk 2]
```diff
@@ -75,7 +75,7 @@
       "value": "${SERVICE_WEB_IDENTITY_NAME}"
     },
     "useApplicationInsights": {
-      "value": "${AZURE_USE_APPLICATION_INSIGHTS=false}"
+      "value": "${AZURE_USE_APPLICATION_INSIGHTS=true}"
```
Comment (jongio) [inline infra/main.parameters.json:78]: I think you should just remove the whole use app insights false logic.

File: infra/main.bicep [hunk 3]
```diff
@@ -495,69 +495,69 @@ module openAiRoleFunction 'core/security/role.bicep' = {
   scope: openAiResourceGroup
   name: 'openai-role-function'
   params: {
-    principalId: principalId
+    principalId: function.outputs.SERVICE_FUNCTION_IDENTITY_PRINCIPAL_ID
```
Comment (jongio) [inline infra/main.bicep:498]: This is correct

**Comment by luisquintanilla** [comment]
@LittleLittleCloud can you please address some of the comments from @jongio before merging.

- Remove app insights logic
- follow SERVICE_FUNCTION pattern in name

Thanks

**Review by luisquintanilla — APPROVED** [review]
Thanks @LittleLittleCloud and @jongio.

**Pull request merged**